### PR TITLE
[Fix] `ES2021+`: values that truncate to -0 in `ToIntegerOrInfinity`

### PR DESCRIPTION
**Pull request by ptomato**
ToIntegerOrInfinity(-0.1) should return 0, not -0.

This was also the case in ES2020's ToInteger (but not in earlier versions) so this adds a similar test to ES2020's tests as well, although no fix is needed there.

### CODE REVIEW

File: 2021/ToIntegerOrInfinity.js [hunk 1]
```diff
@@ -14,5 +14,7 @@ module.exports = function ToIntegerOrInfinity(value) {
 	var number = ToNumber(value);
 	if ($isNaN(number) || number === 0) { return 0; }
 	if (!$isFinite(number)) { return number; }
-	return $sign(number) * floor(abs(number));
+	var integer = floor(abs(number));
+	if (integer === 0) { return 0; }
+	return $sign(number) * integer;
```
Comment (ljharb) [inline 2021/ToIntegerOrInfinity.js:19]: hmm, i'm confused, if `number` is -0, then `sign(number)` would be -0, and `-0 * -0` is `+0`, no?

Comment (ljharb) [reply 2021/ToIntegerOrInfinity.js:19]: ahh, even tho `Math.sign(-0)` is `-0`, my `$sign` helper returns `-1` here. got it.

**Review by ljharb — APPROVED** [review]
Thanks!

File: 2021/ToIntegerOrInfinity.js (hunk 1 above)
Comment (ptomato) [reply 2021/ToIntegerOrInfinity.js:19]: Yep, you beat me to it

**Comment by ljharb** [comment]
v1.20.4 released.

**Pull request merged**